core/controllers/reader.py

```diff
@@ -304,14 +304,33 @@ def post(self, exploration_id):
         client_time_spent_in_secs = self.payload.get(  # pylint: disable=unused-variable
             'client_time_spent_in_secs')
         old_params = self.payload.get('old_params')
+        is_first_hit = self.payload.get('is_first_hit')
 
         # Record the state hit, if it is not the END state.
         if new_state_name is not None:
             event_services.StateHitEventHandler.record(
                 exploration_id, exploration_version, new_state_name,
-                session_id, old_params, feconf.PLAY_TYPE_NORMAL)
+                session_id, old_params, feconf.PLAY_TYPE_NORMAL, is_first_hit)
         else:
             logging.error('Unexpected StateHit event for the END state.')
+        self.render_json({})
+
+
+class StateCompleteEventHandler(base.BaseHandler):
+    """Tracks a learner complete a state. Here, 'completing' means answering
+    the state and progressing to a new state."""
```

Comment:
> Here and elsewhere, when a multi-line docstring ends with """, move the """ to the next line.

I don't think you addressed the "elsewhere" bit. Please double-check your PR.

Reply:
Done.